internal/commands/output.go

```diff
@@ -377,3 +384,67 @@ func (t *TableOutputManager) Flush() error {
 
 	return nil
 }
+
+// JUnitOutputManager formats its output as a JUnit test result
+type JUnitOutputManager struct {
+	p      parser.Package
+	writer io.Writer
+}
+
+// NewDefaultJUnitOutputManager creates a new JUnitOutputManager using standard out
+func NewDefaultJUnitOutputManager() *JUnitOutputManager {
+	return NewJUnitOutputManager(os.Stdout)
+}
+
+// NewJUnitOutputManager creates a new JUnitOutputManager with a given Writer
+func NewJUnitOutputManager(w io.Writer) *JUnitOutputManager {
+	return &JUnitOutputManager{
+		writer: w,
+		p: parser.Package{
+			Name:  "conftest",
+			Tests: []*parser.Test{},
+		},
+	}
+}
+
+// Put puts the result of the check to the manager in the managers buffer
+func (j *JUnitOutputManager) Put(cr CheckResult) error {
+	getOutput := func(r Result) []string {
+		out := []string{
+			r.Message,
+		}
+		for _, err := range r.Traces {
+			out = append(out, err.Error())
+		}
+		return out
+	}
+	convert := func(r Result, status parser.Result) *parser.Test {
+		return &parser.Test{
+			// We have to make sure that the name of the test is unique
+			Name:   fmt.Sprintf("%s - %s", cr.FileName, r.Message),
+			Result: status,
+			Output: getOutput(r),
+		}
+	}
+
+	for _, result := range cr.Warnings {
+		j.p.Tests = append(j.p.Tests, convert(result, parser.FAIL))
```

Comment:
Warnings are a tricky case. I am not sure if we should treat them as failures in JUnit as then they get reported as failures as well (even though that is not the intention), but I also saw nothing in the JUnit spec that would fit.

Reply:
This is the best compromise I found: since conftest will exit with 0 if there is only warnings the test will be considered as successful by the CI which is what we want. The report thought will show those tests and their description.